sonar-orchestrator/src/main/java/com/sonar/orchestrator/server/ServerInstaller.java

```diff
@@ -44,24 +49,29 @@
 
   private static final AtomicInteger sharedDirId = new AtomicInteger(0);
   private final ServerZipFinder zipFinder;
-  private final FileSystem fs;
+  private final Configuration configuration;
   private final DatabaseClient databaseClient;
 
-  public ServerInstaller(ServerZipFinder zipFinder, FileSystem fs, DatabaseClient databaseClient) {
+  public ServerInstaller(ServerZipFinder zipFinder, Configuration configuration, DatabaseClient databaseClient) {
     this.zipFinder = zipFinder;
-    this.fs = fs;
+    this.configuration = configuration;
     this.databaseClient = databaseClient;
   }
 
   public Server install(SonarDistribution distrib) {
-    File home = locateAndUnzip(distrib);
-    configureHome(distrib, home);
-    return new Server(fs, home, distrib);
+    File homeDir = locateAndUnzip(distrib);
+    copyPlugins(distrib, homeDir);
+    copyJdbcDriver(homeDir);
+    Properties properties = configureProperties(distrib);
+    writePropertiesFile(properties, homeDir);
+    Server server = new Server(configuration.fileSystem(), homeDir, distrib);
+    server.setUrl("http://" + properties.getProperty("sonar.web.host") + ":" + properties.getProperty("sonar.web.port") + properties.getProperty("sonar.web.context"));
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "sonar.web.host" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "sonar.web.port" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)